src/com/oltpbenchmark/WorkloadState.java

```diff
@@ -1,126 +1,111 @@
+            if (resetQueues) {
                 workQueue.clear();
-    
-            assert amount > 0;
-    
+            }
+
+
             // Only use the work queue if the phase is enabled and rate limited.
-            if (traceReader != null && currentPhase != null) {
-                if (benchmarkState.getState() != State.WARMUP) {
-                    workQueue.addAll(traceReader.getProcedures(System.nanoTime()));
-               }
-                   }
-            else if (currentPhase == null || currentPhase.isDisabled()
-                || !currentPhase.isRateLimited() || currentPhase.isSerial())
-            {
+            if (currentPhase == null || currentPhase.isDisabled()
+                    || !currentPhase.isRateLimited() || currentPhase.isSerial()) {
                 return;
-                   }
-            else {
+            } else {
                 // Add the specified number of procedures to the end of the queue.
-                for (int i = 0; i < amount; ++i)
+                for (int i = 0; i < amount; ++i) {
                     workQueue.add(new SubmittedProcedure(currentPhase.chooseTransaction()));
-               }
+                }
+            }
 
             // Can't keep up with current rate? Remove the oldest transactions
             // (from the front of the queue).
-            while(workQueue.size() > RATE_QUEUE_LIMIT)
+            while (workQueue.size() > RATE_QUEUE_LIMIT) {
                 workQueue.remove();
+            }
 
             // Wake up sleeping workers to deal with the new work.
-            int numToWake = (amount <= workersWaiting? amount : workersWaiting);
-            for (int i = 0; i < numToWake; ++i)
+            int numToWake = Math.min(amount, workersWaiting);
+            for (int i = 0; i < numToWake; ++i) {
                 this.notify();
-           }
-       }
+            }
+        }
+    }
 
-    public boolean getScriptPhaseComplete() {
-        assert (traceReader != null);
-        synchronized(this) {
-            return traceReader.getPhaseComplete() && workQueue.size() == 0 && workersWorking == 0;
+    public void signalDone() {
+        int current = this.benchmarkState.signalDone();
+        if (current == 0) {
+            synchronized (this) {
+                if (workersWaiting > 0) {
+                    this.notifyAll();
+                }
+            }
         }
-   }
-   
-   public void signalDone() {
-       int current = this.benchmarkState.signalDone();
-       if (current == 0) {
-           synchronized (this) {
-               if (workersWaiting > 0) {
-                   this.notifyAll();
-               }
-           }
-       }
-   }
-   
-   /** Called by ThreadPoolThreads when waiting for work. */
-    public SubmittedProcedure fetchWork(int workerId) {
-        synchronized(this) {
+    }
+
+    /**
+     * Called by ThreadPoolThreads when waiting for work.
+     */
+    public SubmittedProcedure fetchWork() {
```

Comment:
TODO No longer parameterized by workerId, not that I think it makes a huge diff?

Reply:
Added to discussion.